vivisect/analysis/generic/pointers.py

```diff
@@ -62,30 +64,45 @@ def analyze(vw):
         try:
             logger.debug('pointer(4): 0x%x -> 0x%x', addr, pval)
             vw.makePointer(addr, follow=True)
-            done.append((addr, pval))
+            done[addr] = pval
         except Exception as e:
             logger.error('makePointer() failed for 0x%.8x (pval: 0x%.8x) (err: %s)', addr, pval, e)
 
-    # Now let's see what these guys should be named (if anything)
-    for _, _ in done:
-        for ptr, tgt in done:
-            try:
-                pname = vw.getName(ptr)
-                if pname is not None:
-                    logger.debug('skipping renaming of ptr 0x%x (currently: %r)', ptr, pname)
```

Comment:
hmmm... so i like your design and implementation... 
however, you have zero logging, which means that Pointers get to be made willy-nilly and we never know why or why not, and when pointer names get overwritten (or not, but collisions happen), we don't know why or what the thing is, or was.

until we get a better handle on Pointers that *shouldn't* be, i'd like some logging about what things are being done and why.

pointer/opcode/etc collisions are one the best indications where codeflow and pointer analysis go awry.  don't take that away.  rather, since this is so noisy, please use the new SHITE log-level.

Reply:
logging collisions is important.  it also shows us when different analysis modules are bumping into each other, since each pointer comes from somewhere, and some of our analysis modules underestimate what others do.

Reply:
More logging is fine. Though after our conversations, I doubt my version of more logging matches up to your version. But I will add more.

Also, you new SHITE logging isn't merged in. Just as heads up that won't work quiiiiite yet.

Reply:
heh.  yeah... it's only three lines.  feel free to add them.

```
--- a/envi/common.py
+++ b/envi/common.py
@@ -4,6 +4,7 @@ LOG_FORMAT = '%(asctime)s:%(levelname)s:%(name)s: %(message)s' \
              '[%(filename)s:%(funcName)s:%(lineno)s:%(threadName)s]'
 
 EMULOG = 9
+SHITE = 8
 
 LOG_LEVELS = (
     logging.CRITICAL,
@@ -12,6 +13,7 @@ LOG_LEVELS = (
     logging.INFO,
     logging.DEBUG,
     EMULOG,
+    SHITE,
 )
 
 def initLogging(logger, level=None, fmt=LOG_FORMAT):
@@ -26,6 +28,7 @@ def initLogging(logger, level=None, fmt=LOG_FORMAT):
 
     # Setup logging
     logging.addLevelName(EMULOG, 'EMULOG')
+    logging.addLevelName(SHITE, 'SHITE')
     logging.basicConfig(level=level, format=fmt)
 
     # Set log level for current logger
```

;)

Reply:
I'd like to keep those in their own PR, since I'm not a huuuuge fan of the name. Other levels should suffice for now. I'm not planning to add that many in all honesty. I've found the pointers logging to be both extremely verbose and not entirely helpful.

Reply:
yeah, i wondered why there were *so* many messages coming from pointers.py.  i guess the O-N2 problem would do that.